test/framework/resources/k8s/resources/node.go

```diff
@@ -38,6 +39,9 @@ func NewDefaultNodeManager(k8sClient client.DelegatingClient) NodeManager {
 }
 
 func (d *defaultNodeManager) GetNodes(nodeLabelKey string, nodeLabelVal string) (v1.NodeList, error) {
+	if nodeLabelVal == "" {
```

Comment:
Added this because, the test would fail if nodeLabelVal is not specified as No nodes would be returned. 
We can either make this mandatory field or return all nodes if user hasn't specified any filtering criteria